### ISSUE TITLE
Server claims gzip but sends uncompressed; curl not like

### ISSUE THREAD

**Issue by bagder** [curl/curl#2368]
### I did this

```
$ curl --compressed https://www.ica.se/
curl: (61) Error while processing content unencoding: incorrect header check
```
It seems to happen because the server sends back a `Content-Encoding: gzip` header although the data doesn't seem to be compressed at all.

curl 7.57.0 manages to display this fine.

### I expected the following

To see the HTML contents

### curl/libcurl version

7.58.0 and current master

### operating system

All

**Comment by monnerat** [comment]
Can't reproduce here. Maybe because of what is compiled in (tell me your features, please).
Anyway, if the server tells the data is gzip-encoded while it's not, it is a server error and curl reacts well.

**Comment by monnerat** [comment]
I've just simulated such a lame server with the following PHP script:
```
<?php
  header('Content-Encoding: gzip');
?>
<html>
<body>
<p>This is a paragraph.</p>
</body>
</html>
```

I reproduce the error with master. However 7.57.0 says
`curl: (23) Error while processing content unencoding: invalid code lengths set`
because it tried to fallback to the deflate algorithm without headers (old M$ servers deflate can be lame in this way too :-/ ).

I've tried back to 7.55.0, but I can't find a version that does not terminate in error.
